test/e2e/framework/profile_gatherer.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+)
+
+func GatherApiserverCPUProfile(testBaseName string) error {
+	sshResult, err := SSH("wget -O - localhost:8080/debug/pprof/profile", GetMasterHost()+":22", TestContext.Provider)
+	if err != nil {
+		return fmt.Errorf("Failed to execute wget command on master through SSH: %v", err)
+	}
+	filePath := path.Join(TestContext.ReportDir, "ApiserverCPUProfile_"+testBaseName+".prof")
+	err = ioutil.WriteFile(filePath, []byte(sshResult.Stdout), 0644)
+	if err != nil {
+		return fmt.Errorf("Failed to write file %v with profile data: %v", filePath, err)
+	}
+	return nil
+}
+
+func GatherApiserverMemoryProfile(testBaseName string) error {
+	sshResult, err := SSH("wget -O - localhost:8080/debug/pprof/heap", GetMasterHost()+":22", TestContext.Provider)
```

Comment:
IIUC this is basically heap size.
Can we also have the profile of all allocations since the beginning of the test?

Reply:
IIUC the data which is obtained by the above command contains both inuse and alloc info of the heap.
If we later want to graph the allocations data, we can get it from this using `go tool pprof --alloc-space`.
One thing I'm still not 100% sure about is if the allocations data would be since the beginning of the binary (maybe you have an answer to it?)